include/psen_scan_v2/reply_msg_from_scanner.h

```diff
@@ -41,23 +41,30 @@ class ReplyMsgFromScanner
 public:
   static ReplyMsgFromScanner fromRawData(const RawScannerData& data);
 
-public:
-  ReplyMsgFromScanner(const uint32_t op_code, const uint32_t res_code);
+private:
+  template <typename T>
+  static void read(std::istringstream& is, const T& data);
 
 public:
+  ReplyMsgFromScanner(const uint32_t op_code, const uint32_t res_code);
   ReplyMsgFromScannerType type() const;
-
-public:
   static uint32_t getStartOpCode();
   static uint32_t calcCRC(const ReplyMsgFromScanner& msg);
 
+private:
+  template <typename T>
+  static void process_bytes(boost::crc_32_type& crc_32, const T& data);
+
+public:
   using RawType = std::array<char, REPLY_MSG_FROM_SCANNER_SIZE>;
   RawType toCharArray();
 
 private:
+  template <typename T>
+  void write(std::ostringstream& os, const T& data) const;
```

Comment:
I noticed that the `write()` method is basically the same in the `ReplyMsgFromScanner` class and in the `StartRequest` class. Therefore, I suggest to separate this function from the before mentioned classes and, consequently, avoid code duplication.